cpp/src/arrow/flight/client.h

```diff
@@ -86,20 +92,21 @@ class ARROW_EXPORT FlightClient {
 
   /// \brief Given a flight ticket and schema, request to be sent the
   /// stream. Returns record batch stream reader
-  /// \param[in] ticket
+  /// \param[in] ticket The flight ticket to use
   /// \param[in] schema the schema of the stream data as computed by
   /// GetFlightInfo
   /// \param[out] stream the returned RecordBatchReader
   /// \return Status
   Status DoGet(const Ticket& ticket, const std::shared_ptr<Schema>& schema,
                std::unique_ptr<RecordBatchReader>* stream);
 
-  /// \brief Initiate DoPut RPC, returns FlightPutWriter interface to
-  /// write. Not yet implemented
-  /// \param[in] schema the schema of the stream data
-  /// \param[out] stream the created stream to write record batches to
+  /// \brief Upload data to a Flight described by the given descriptor.
+  /// \param[in] descriptor the descriptor of the stream
+  /// \param[in] schema the schema for the data to upload
+  /// \param[out] stream a writer to write record batches to
```

Comment:
Comment that the caller is responsible for calling `Close` on the returned object to end the Put stream